src/main/java/de/unistuttgart/overworldbackend/service/CourseService.java

```diff
@@ -128,11 +136,11 @@ private void configureWorld(List<World> worlds, int worldId, WorldConfig worldCo
   private Dungeon configureDungeon(int dungeonId, DungeonConfig dungeonConfig) {
     Set<MinigameTask> minigames = new HashSet<>();
     Set<NPC> npcs = new HashSet<>();
-    for (int minigameIndex = 0; minigameIndex < dungeonConfig.getNumberOfMinigames(); minigameIndex++) {
+    for (int minigameIndex = 1; minigameIndex <= dungeonConfig.getNumberOfMinigames(); minigameIndex++) {
       MinigameTask minigame = new MinigameTask(null, null, minigameIndex);
       minigames.add(minigame);
     }
-    for (int npcIndex = 0; npcIndex < dungeonConfig.getNumberOfNPCs(); npcIndex++) {
+    for (int npcIndex = 1; npcIndex <= dungeonConfig.getNumberOfNPCs(); npcIndex++) {
       NPC npc = new NPC("", npcIndex);
       npcs.add(npc);
     }
```

Comment:
I just noticed that this method can also be simplified to a theoretical one-liner:
```java
return new Dungeon(dungeonConfig.getStaticName(), "", false, IntStream.rangeClosed(1, dungeonConfig.getNumberOfMinigames()).map(i -> new MinigameTask(null, null, i)).collect(Collectors.toCollection(HashSet::new)), IntStream.rangeClosed(1, dungeonConfig.getNumberOfNPCs()).map(i -> new NPC("", i)).collect(Collectors.toCollection(HashSet::new)), dungeonId);
```

Reply:
Nice, but I will not implement this here in this PR. Also, I do not really know if this improves readability.

Reply:
Me neither...